### PR TITLE
fix: Add plumbing for gecko build system integration

### PR DESCRIPTION
**Pull request by larseggert**

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -4,7 +4,38 @@
...
+fn clang_args() -> Vec<String> {
+    use mozbuild::TOPOBJDIR;
+
+    let flags_path = TOPOBJDIR.join("netwerk/socket/neqo/extra-bindgen-flags");
+    println!("cargo:rerun-if-changed={}", flags_path.to_str().unwrap());
+
+    let mut flags: Vec<String> = std::fs::read_to_string(flags_path)
+        .expect("Failed to read extra-bindgen-flags file")
+        .split_whitespace()
+        .map(std::borrow::ToOwned::to_owned)
```
Comment (mxinden) [inline build.rs:19]: Can you call `to_owned` directly on the `Iterator` returned by `split_whitespace()`?

```suggestion
        .to_owned()
```

https://doc.rust-lang.org/stable/std/str/struct.SplitWhitespace.html#impl-ToOwned-for-T

**Review by mxinden — APPROVED** [review]
For what my limited knowledge of the Mozilla build system is worth, this looks good to me.

File: build.rs (hunk 1 above)
Comment (larseggert) [reply build.rs]: This is used in a bunch of other places in gecko; might want to do a PR over there, too.

Comment (mxinden) [reply build.rs]: If you feel strongly about it, let me know, happy to do it @larseggert.

If not, I would argue the overhead of producing a mozilla-central patch is not worth the small cleanup.

**Pull request merged**